src/components/SolidLoginForm.jsx

```diff
@@ -1,38 +1,63 @@
 import { getDefaultSession, fetch } from "@inrupt/solid-client-authn-browser";
-import { getLiteral, getProfileAll, getThing } from "@inrupt/solid-client";
+import {
+  getLiteral,
+  getProfileAll,
+  getThing,
+} from "@inrupt/solid-client";
 import { FOAF } from "@inrupt/vocab-common-rdf";
 import { useState } from "react";
+import { QueryEngine } from "@comunica/query-sparql";
 
 function SolidLoginForm(props) {
   const session = getDefaultSession();
   let webId = session.info.webId;
   const [name, setName] = useState(undefined);
+  const [errorMessage, setErrorMessage] = useState(undefined);
+
   if (webId) {
-    getProfileAll(webId, { fetch: fetch }).then((dataSet) => {
-      let profile = dataSet.webIdProfile;
-      const webIdThing = getThing(profile, webId);
-      let literalName = getLiteral(webIdThing, FOAF.name);
-      if (literalName) {
-        setName(literalName.value);
-      } else {
-        setName(webId);
-      }
-    }).catch(_ => setName(webId));
+    getProfileAll(webId, { fetch: fetch })
+      .then((dataSet) => {
```

Comment:
Can we use async/await here or is this also against the best practices of React to do that here?

Reply:
We would have to make the function, which is a React component, async, but than it would return a promise and not a React component. I have worked around this however by defining the async function in an effect which is better here either way. This is done in [commit fe7c3ff](https://github.com/SolidLabResearch/generic-data-viewer/pull/52/commits/fe7c3ff00b67f63182ea5ef1f9e305ceb4d102f8)